src/main/java/software/amazon/event/kafkaconnector/batch/BestEffortInOrderEventBridgeBatching.java

```diff
@@ -0,0 +1,90 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+package software.amazon.event.kafkaconnector.batch;
+
+import static java.nio.charset.StandardCharsets.UTF_8;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.stream.Collector;
+import java.util.stream.Stream;
+import software.amazon.awssdk.services.eventbridge.model.PutEventsRequestEntry;
+import software.amazon.event.kafkaconnector.util.MappedSinkRecord;
+
+public class BestEffortInOrderEventBridgeBatching implements EventBridgeBatchingStrategy {
+
+  private final Collector<
```

Comment:
Q: What is the advantage of this over PutEventsRequestEntrySizeCollector implementing the Collector interface?

Reply:
To implement the `Collector` interface we still need an object for the state. It would looks like:
```
    static class Accumulator {

        private static final int MAX_BATCH_SIZE_BYTES = 256 * 1024;

        private final List<List<MappedSinkRecord<PutEventsRequestEntry>>> result = new ArrayList<>();
        private int actualBatchSize = 0;
        private int index = 0;

       ...        
    }

    static class PutEventsRequestEntrySizeCollector implements Collector<MappedSinkRecord<PutEventsRequestEntry>,
            Accumulator,
            Stream<List<MappedSinkRecord<PutEventsRequestEntry>>>> {

        @Override
        public Supplier<Accumulator> supplier() { ... }

        @Override
        public BiConsumer<Accumulator, MappedSinkRecord<PutEventsRequestEntry>> accumulator() { ... }

        @Override
        public BinaryOperator<Accumulator> combiner() { ... }

        @Override
        public Function<Accumulator, Stream<List<MappedSinkRecord<PutEventsRequestEntry>>>> finisher() { ... }

        @Override
        public Set<Characteristics> characteristics() { ... }
    }
```
If it's better then I'm fine.

Reply:
Yes I think that makes it a bit more explicit.

Reply:
Agree. Will change it.